File: frontend/types/interview.ts
import type { Application, InterviewMode, InterviewResult } from "@/types/application";

export type { InterviewMode, InterviewResult };

export interface Interview {
  id: number;
  applicationId: number;
  userId?: number;
  roundName: string;
  interviewDate: string;
  mode: InterviewMode;
  result: InterviewResult;
  notes?: string;
  createdAt: string;
}

export interface InterviewPayload {
  roundName: string;
  interviewDate: string;
  mode: InterviewMode;
  result?: InterviewResult;
  notes?: string;
}

export interface InterviewWithApplication extends Interview {
  application?: Application;
  companyName?: string;
  role?: string;
}

export interface InterviewFilters {
  search: string;
  result: InterviewResult | "";
  mode: InterviewMode | "";
}

File: backend/src/main/java/com/jobflow/interviews/entity/Interview.java
package com.jobflow.interviews.entity;

import com.jobflow.applications.entity.Application;
import com.jobflow.interviews.enums.InterviewMode;
import com.jobflow.interviews.enums.InterviewResult;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "interviews", indexes = {
    @Index(name = "idx_interviews_application_id", columnList = "application_id"),
    @Index(name = "idx_interviews_interview_date", columnList = "interview_date"),
    @Index(name = "idx_interviews_result", columnList = "result")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Interview {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "application_id", nullable = false)
    private Application application;

    @Column(name = "round_name", nullable = false, length = 150)
    private String roundName;

    @Column(name = "interview_date", nullable = false)
    private LocalDateTime interviewDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private InterviewMode mode;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    @Builder.Default
    private InterviewResult result = InterviewResult.PENDING;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}
